firmware/quadruna/BMS/src/app/app_diagnosticsMode.c

```diff
@@ -0,0 +1,253 @@
+#include "app_diagnosticsMode.h"
+#include "states/app_chargeState.h"
+#include "app_canTx.h"
+#include "app_canRx.h"
+#include "ltc6813/io_ltc6813Shared.h"
+#include "ltc6813/io_ltc6813CellVoltages.h"
+#include "ltc6813/io_ltc6813CellTemps.h"
+
+#include <string.h>
+#include <float.h>
+
+typedef struct
+{
+    float   cell_voltages[ACCUMULATOR_NUM_SEGMENTS][ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT];
+    uint8_t segment_temps[ACCUMULATOR_NUM_SEGMENTS];
+} BMSDiagnosticStats;
+
+static BMSDiagnosticStats data;
+
+void app_diagnosticsMode_calculateDiagnosticVoltageStats(void)
+{
+    // Find the min and max voltages
+    for (uint8_t segment = 0U; segment < ACCUMULATOR_NUM_SEGMENTS; segment++)
+    {
+        for (uint8_t cell = 0U; cell < ACCUMULATOR_NUM_SERIES_CELLS_PER_SEGMENT; cell++)
+        {
+            // Collect each cell voltage and store in temp
+            const float cell_voltage          = io_ltc6813CellVoltages_getCellVoltage(segment, cell);
+            data.cell_voltages[segment][cell] = cell_voltage;
+        }
+    }
+}
+
+void app_diagnosticsMode_calculateDiagnosticTemperatureStats(void)
+{
+    for (uint8_t curr_segment = 0U; curr_segment < ACCUMULATOR_NUM_SEGMENTS; curr_segment++)
+    {
+        float sum_segment_temp = 0U;
+
+        for (uint8_t curr_reg_group = 0U; curr_reg_group < NUM_OF_AUX_REGISTER_GROUPS; curr_reg_group++)
+        {
+            for (uint8_t curr_thermistor = 0U; curr_thermistor < NUM_OF_READINGS_PER_REG_GROUP; curr_thermistor++)
+            {
+                if (IS_CELL_TEMP_READING(curr_reg_group, curr_thermistor))
+                {
+                    const float curr_cell_temp = io_ltc6813CellTemperatures_getSpecificCellTempDegC(
+                        curr_segment, curr_reg_group, curr_thermistor);
+                    sum_segment_temp += curr_cell_temp;
+                }
+            }
+        }
+
+        data.segment_temps[curr_segment] = (uint8_t)(sum_segment_temp / NUM_OF_THERMISTORS_PER_SEGMENT);
+    }
+}
+
+void app_diagnosticsMode_broadcast(void)
+{
+    app_diagnosticsMode_calculateDiagnosticVoltageStats();
```

Comment:
```suggestion
bool cellVoltageDiagnosticEnabled = app_canRx_Debug_CellVoltageDiagnoticEnabled_get();
if( cellVoltageDiagnoticEnabled){
    app_diagnosticsMode_calculateDiagnosticVoltageStats();
    }
```
Only calculate the Diagnotic stats if the the diagnotic mode is enabled to make it more efficient in normal operation.
Instead add a bool for cell voltage diagnostic mode to the CAN message to enable or disable the stats calculation

Its already this way for Temp Diagnostic Mode.